### DISCUSSION TITLE
Compile Lua-Scripts in Windows

### DISCUSSION THREAD

**Discussion by Andreas0168** [EdgeTX/edgetx#4896] [LUA]
Hello,

I'm currently building a LUA script for the MT12 and the script has already become very large. Now I often have the problem that I get the error message 'not enough memory' after I have loaded the script with new ideas onto the SD card. In order to compile the new script I have to switch to a model without any settings so that it works.
Is there a tool for Windows with which I can compile LUA scripts into LUAC suitable for EdgeTX? I discovered that actually only the LUAC is needed in EdgeTX.
I don't necessarily want to have to split my script into several scripts.

**Comment by ChrisOhara57** [comment]
Hello
I'm still using OpenTx and Companion but I do recon it's the same in EdgeTx.
Use the Simulator in Companion instead of the real Radio.
There is no run-time limitation while compiling the LUA into LUAC.
And then you can copy the LUA and LUAC to your radio.
(I think INAV is n example of a scripts  that has to be compiled this way.)
I think they also recommend to rename the .LUAC file to .LUA (possible to prevent the radio to compile a second time)
